File: src-tauri/src/commands/env.rs
use crate::models::result::CommandResult;
use crate::services::env_service::{self, DetectEnvData};

#[tauri::command]
pub async fn detect_env() -> CommandResult<DetectEnvData> {
    match env_service::detect_env().await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

File: src/main/ipc/env.ts
import { ipcMain } from "electron";
import { IPC } from "../../shared/ipc-channels.js";
import { ok, err } from "../models/result.js";
import { toAppError } from "./utils.js";
import { detectEnv } from "../services/env-service.js";

export function registerEnvHandlers(): void {
  ipcMain.handle(IPC.DETECT_ENV, async () => {
    try { return ok(await detectEnv()); }
    catch (error) { return err(toAppError(error)); }
  });
}
